src/components/Input/Input.tsx

```diff
@@ -0,0 +1,48 @@
+import classNames from 'classnames';
+import { useState, useEffect, forwardRef } from 'react';
+
+import styles from './Input.module.scss';
+
+type InputProps = Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange' | 'value'> & {
+  /** Значение поля */
+  value: string;
+  /** Callback, вызываемый при вводе данных в поле */
+  onChange: (value: string) => void;
+  /** Слот для иконки справа */
+  afterSlot?: React.ReactNode;
+};
+
+const Input = forwardRef<HTMLInputElement, InputProps>(
+  ({ value, onChange, afterSlot, type = 'text', className, ...props }, ref) => {
+    const [inputValue, setInputValue] = useState(value);
+
+    useEffect(() => {
+      setInputValue(value);
+    }, [value]);
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      setInputValue(e.target.value);
+      onChange(e.target.value);
+    };
```

Comment:
зачем тебе внутреннее состояние со строкой инпута? просто прокинь value сразу в input value
```suggestion
    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
      onChange(e.target.value);
    };
```

Reply:
Убрал внутреннее состояние из компонента Input